frontend/src/metabase/lib/utils.unit.spec.ts

```diff
@@ -0,0 +1,105 @@
+import MetabaseUtils from "metabase/lib/utils";
+
+describe("utils", () => {
+  describe("versionToNumericComponents", () => {
+    it("should pad to 4 numeric components", () => {
+      expect(
+        MetabaseUtils.versionToNumericComponents("v1.2-BETA1"),
+      ).toStrictEqual([1, 2, 0, 0, -2, 1]);
+    });
+  });
+  describe("compareVersions", () => {
+    it("should compare versions correctly", () => {
+      const expected = [
+        "0.0.9",
+        "0.0.10-snapshot",
+        "0.0.10-alpha1",
+        "0.0.10-rc1",
+        "0.0.10-rc2",
+        "0.0.10-rc10",
+        "0.0.10",
+        "0.1.0",
+        "0.2.0",
+        "0.10.0",
+        "1.1.0",
+      ];
+      const shuffled = expected.slice();
+      shuffle(shuffled);
+      shuffled.sort(MetabaseUtils.compareVersions);
```

Comment:
:x: Actually, this shuffling does not work, rendering this test useless.
`Array.prototype.sort` returns a new array.

```suggestion
      const shuffled = shuffle(expected).sort(MetabaseUtils.compareVersions);
```

Reply:
Array.prototype.sort sorts in place and returns a reference to the array itself: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort
<img width="255" alt="image" src="https://github.com/metabase/metabase/assets/1914270/15bbeb1a-2d77-45d6-8b42-ddb20f1e2013">

Am I missing something?

Reply:
You're right.
I need one more coffee.

Nonetheless, if [this suggestion](https://github.com/metabase/metabase/pull/33671#discussion_r1315865663) is applied, then this suggestion also applies.